client/rest-high-level/src/main/java/org/elasticsearch/client/AsyncSearchRequestConverters.java

```diff
@@ -56,4 +55,21 @@ static Request submitAsyncSearch(SubmitAsyncSearchRequest asyncSearchRequest) th
         request.addParameters(params.asMap());
         return request;
     }
+
+    static void addSearchRequestParams(Params params, SubmitAsyncSearchRequest request) {
+        params.putParam(RestSearchAction.TYPED_KEYS_PARAM, "true");
+        params.withRouting(request.getRouting());
+        params.withPreference(request.getPreference());
+        params.withIndicesOptions(request.getIndicesOptions());
+        params.putParam("search_type", request.getSearchType().name().toLowerCase(Locale.ROOT));
+        if (request.getRequestCache() != null) {
+            params.putParam("request_cache", Boolean.toString(request.getRequestCache()));
+        }
+        if (request.getAllowPartialSearchResults() != null) {
+            params.putParam("allow_partial_search_results", Boolean.toString(request.getAllowPartialSearchResults()));
+        }
+        params.putParam("batched_reduce_size", Integer.toString(request.getBatchedReduceSize()));
+        params.putParam("pre_filter_shard_size", Integer.toString(SubmitAsyncSearchRequest.DEFAULT_PRE_FILTER_SHARD_SIZE));
+        params.putParam("ccs_minimize_roundtrips", Boolean.toString(SubmitAsyncSearchRequest.DEFAULT_CCS_MINIMIZE_ROUNDTRIPS));
```

Comment:
should this be there given that we don't allow to set it directly and we support only one value for it?

Reply:
Yes, wee need to send those values (the onse we set on the ctor) via HTTP because they are different than the defaults in SearchRequest.

Reply:
Okay, upon rechecking the code in RestSearchAction#prepareRequest, we set those values in the ctor on the server side to the defaults and shold only send those that the user can still modify, like `batchedReduceSize` and `requestCache`(see https://github.com/elastic/elasticsearch/pull/53592#discussion_r393863122)

Reply:
Removing "pre_filter_shard_size" and "ccs_minimize_roundtrips" from the http serialization